File: src/2942-Find-Words-Containing-Character/find_words_containing_character.cpp
#include <iostream>
#include <vector>
#include <string>

class Solution {
public:
    std::vector<int> findWordsContaining(std::vector<std::string>& words, char x) {
        std::vector<int> ans = {};
        int i = 0;

        for (const std::string& w : words) {
            if (w.find(x) != std::string::npos) {
                ans.push_back(i);
            }
            i += 1;
        }

        return ans;
    }
};

int main() {
    std::vector<std::string> words = {"leet","code"};
    char x = 'e';
    
    Solution solution;

    std::vector<int> output = solution.findWordsContaining(words, x);

    for (auto it : output) {
        std::cout << it << " ";
    }
    
    return 0;
}

File: src/2942-Find-Words-Containing-Character/find_words_containing_character.py
from typing import List

class Solution:
    def findWordsContaining(self, words: List[str], x: str) -> List[int]:
        ans: List[int] = []
        i: int = 0

        for w in words:
            if x in w:
                ans.append(i)
            i += 1

        return ans


if __name__ == "__main__":
    words: List[str] = ["leet","code"]
    x: str = "e"
    solution = Solution()
    
    print(solution.findWordsContaining(words, x))